src/backend/distributed/commands/cluster.c

```diff
@@ -69,6 +70,20 @@ PreprocessClusterStmt(Node *node, const char *clusterCommand,
 		return NIL;
 	}
 
+	/* we do not support CLUSTER command on partitioned tables as it can not be run inside transaction blocks */
+	if (PartitionedTableNoLock(relationId))
```

Comment:
why do you prefer `PartitionedTableNoLock` over `PartitionedTable`? We seem to use the former for performance critical code-paths, which seems irrelevant here.

Reply:
Several lines earlier, we already got an AccessExclusive lock. I assumed we no longer need to acquire another lock. I believe it is highly unlikely that we stop getting that lock there in the future.

https://github.com/citusdata/citus/blob/b79111527e70fff309f9df3c9067620168bc7605/src/backend/distributed/commands/cluster.c#L52-L54